### DISCUSSION TITLE
Integrating a camera with Crazyflie?

### DISCUSSION THREAD

**Discussion by shreeram-murali** [bitcraze/discussions#419] [Q&A] [answered]
Hi,

I’ve been trying to figure out how to get images from a camera onboard the Crazyflie. I’d like to have the camera downward facing and I’d want to stream images at 30 fps (preferably 60 fps). I’m thinking of a Raspberry PiCam + Pi0 [(as written about here)](https://www.bitcraze.io/2017/07/crazyflie-based-quadcopter-with-raspberrypi-camera/) or something much smaller and lighter — ESP32 + ESP32cam? 

I had previously asked about how to try and achieve this using the AI deck (in #127 and #265) but was pointed out that there’s a bottleneck in the GAP8 firmware that caps the fps at around 7-12 fps. 

Any hints would be appreciated. Thanks in advance!

**Comment by ataffanel** [comment, marked as answer]
Using an ESP32 we have managed to achieve 30fps, though the camera itself could compress in JPEG which is what makes it possible. It would most likely not be possible if the camera only outputs raw RGB (or worse the bayer image like what we get on the AIdeck color camera).

The raspberrypi zero w would most likely yield much better results: using the hardware h264 encoding there should be no problem streaming 60fps over wifi, though flying a raspberrypi zero with the Crazyflie 2.1 is a challenge, it barely fly. Maybe the new upcoming [motor update kit](https://www.bitcraze.io/2022/10/thrust-upgrade-kit-for-the-crazyflie-2-1/) could help but that has not been tested yet.

**Reply by shreeram-murali** [reply]
Hi, thanks for the reply. What was the camera module used (that could compress to JPEG)?

**Reply by knmcguire** [reply]
Hi! Please make a new discussion for new questions.
